### DISCUSSION TITLE
v3.2.3

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#4010] [Releases]
## Installation
**iOS**: Visit [https://getutm.app/install/][1] for the most up to date installation instructions.
**macOS**: Download "UTM.dmg" below and copy the contents to your `/Applications` directory.

## Highlights
* **QEMU 7.0.0**: The backend has been updated to the latest release. Check out https://www.qemu.org/2022/04/19/qemu-7-0-0/ for more details. Additionally, other backend components such as SPICE GTK, libusbredir, and more have also been updated.
* **Reclaiming space**: the host will attempt to TRIM any page of all zeros when committed to the QCOW2 disk image. Additionally, a new button in disk settings will allow you to re-convert a disk image to reclaim space (of all-zero pages) as well as compress an existing disk image (macOS 12+ only). Note compression is only done statically which means when the guest modifies a sector, it will be stored uncompressed just like before.
* **Disposable mode**: right click/long press on a QEMU VM and select "Run without saving changes" to start VM in "disposable mode." In this mode, changes will not be saved to disk and will be discarded once the VM is stopped. (thanks @ktprograms)

## Notes
* Reminder: for troubleshooting common Windows install issues (such as no networking), please check out the [guide](https://mac.getutm.app/gallery/windows-10).
* If you manually added `highmem=on` or `highmem=off` to your QEMU Machine Properties as a workaround to issues in a previous release, please remove it.
* (iOS) **UTM v3.2.x will be the last update to support iOS 11-13.** If you are still running those versions, please make sure you report any issues now so they can be fixed by the final release of v3.2.x. After that, iOS 14+ will be required to run future versions of UTM.

## Changes (v3.2.3)
* Disabled screenshots for GL virtual machines (workaround for a crash that occurs, this will be re-enabled in a future update once the crash is triaged)
* (iOS) Pass through Esc key on external keyboard when VoiceOver is in use (#3940)
* (macOS) Use correct scaling when HiDPI is enabled (#3968)
* (macOS) Fix toolbar overlapping top of screen when "Automatically hide and show the menu bar in full screen" is unchecked in system preferences (#3919)
* (macOS) When bridged networking is selected, default the interface to "en0" (#3945)
* (macOS) Wizard: clarify that Linux kernel and initrd must be uncompressed if Apple Virtualization is used

## Changes (v3.2.2)
* Add "disposable mode" (see highlights for more information) (thanks @ktprograms)
* Do not enable UEFI for unsupported architectures (#3976)
* Fix broken play button in sidebar (#3980)
* Fix "Error -1 while loading VM state" on resume caused by usbredir (#3979)
* Various UI tweaks and typo fixes (#3984)
* Fixed pausing without saving state
* Fixed showing warning prompt when VM is paused but not saved
* (iOS) Detect if UTM is installed without AltStore and AltJIT will not attempt to be used.
* (iOS) Reduced build size by 10% (thanks @Halo-Michael) (#3934)
* (macOS) Add new preference option to revert Caps Lock behaviour to pre 3.2.0 (#3974)

## Changes (v3.2.1)
* Fixed boot of RISC-V virt platform (thanks @wegank)
* Remove `highmem=off` default property which is no longer needed on macOS 12.4+ and iOS 15.5+ (it will still be used on lower versions) (#3941, #3946)
* Caps lock state will now be synchronized when mouse is captured (note that if mouse is captured before the guest OS has booted up, the state will be out of sync) (#3415)
* Re-introduce option to disable auto-resolution when SPICE tools is installed (#3938)
* When discarding unsaved settings, do not delete any disk image not created by UTM (#3587)
* Fix downloading of UTM zip files when the URL does not end in ".zip" (#3953)
* (macOS) When importing a disk image, do not silently overwrite an existing image of the same name (#3866)
* (macOS) Added option to reclaim unused space from disk image and to compress a disk image (macOS 12+ only)
* (macOS) On the VM list, the start button will now change to a spinner when the VM is starting and will disappear when the VM has started. Also added the ability to start/show a VM by double-clicking on the list item (#3885)

## Changes (v3.2.0)
* Checkbox for "Import VHDX Image" in the wizard now works correctly again (#3760)
* Fixed various crashes in SPICE webdavd share has been resolved thanks to updated backend (#3633)
* Removed "cache=writethrough" argument from block devices since it was added to workaround an issue with VHDX images which we now convert to QCOW2
* Added "discard=unmap,detect-zeroes=unmap" argument to block devices to reclaim unused space (#3460)
* When "do not save screenshot to disk" is enabled, any existing screenshot will be deleted the next time the VM is started (#3936)
* Fixed an issue in QEMU additional arguments where it sometimes would not appear after setting it (#3895)
* Fixed RNG device not appearing in console mode even when enabled in the settings (#3790)
* Added support for up to 5 additional mouse buttons (you may need to re-map them in the guest OS as they are automatically set to middle/up/down/side/extra) (#3906)
* (iOS) Fixed an issue in the legacy view where "invalid identifier" error may show up on a newly created VM (#3872)
* (macOS) Fixed crash when changing display settings while the VM window is open (but stopped) (#3925)
* (macOS) Fixed crash when trying to select "Share..." in context menu (#3915)
* (macOS) Prevent allowing rapid clicking of pause/resume which may lead to VM corruption (#3629)
* (macOS) Show progress indicator while VM is saving to prevent user from terminating the app early and cause corruption (#3752)
* (macOS) Block unsupported Apple USB devices from being captured (#3759)

## Issues
Please check the full list on [Github][2] and help report any bug you find that is not listed.

### Known Issues

* iOS 14 and below: There is a known issue crashing while running under taurine but not unc0ver. (#3628)

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v3.2.3'>v3.2.3</a>.</em>

**Comment by Andrupka** [comment]
For some reason this build is a lot slower on iPhone 7, with Windows XP SP2 and SPICE tools installed. Even quake is unplayable now, resorting back to 3.2.2 as it's the fastest one yet

**Reply by osy** [reply]
Hmm, could it be an QEMU 7.0.0 issue?

**Reply by Andrupka** [reply]
Maybe, but I think 3.2.2 is also 7.0.0 QEMU based

**Reply by osy** [reply]
QEMU didn’t change and also very little iOS code changed (just some toolbar stuff). Any chance you can test it a few more times to be certain it’s an issue and not a fluke?

**Reply by Andrupka** [reply]
I tested, with the same VM. Boot times are 20 seconds slower, even dragging windows lags. I tested several times
